cns/requestcontroller/kubecontroller/crdtranslator.go

```diff
@@ -57,10 +58,15 @@ func CRDStatusToNCRequest(crdStatus nnc.NodeNetworkConfigStatus) (cns.CreateNetw
 				return ncRequest, fmt.Errorf("Invalid SecondaryIP %s:", ipAssignment.IP)
 			}
 
-			secondaryIPConfig = cns.SecondaryIPConfig{
-				IPAddress: ip.String(),
+			ipConfig, ok := ncRequest.SecondaryIPConfigs[ipAssignment.Name]
+			if !ok || (ok && ipConfig.IPAddress != ip.String()) {
+				ncVersion, _ := strconv.Atoi(ncRequest.Version)
+				secondaryIPConfig = cns.SecondaryIPConfig{
+					IPAddress: ip.String(),
+					NCVersion: ncVersion,
```

Comment:
Is this logic expected? For the existing SecondaryIPs, are we updating the NCversion on CRD update? I thought the whole purpose of adding a version to SecondaryIP was to preserve the version when secondaryip was added to this NC

Reply:
This logic is expected. We are updating the NCversion on CRD update. CRDStatusToNCRequest is the function will be called when CRD update.
CNS don't want to renew existing IPs in secondary IP.